docs/concepts/advanced/rendering_js_css.md

```diff
@@ -220,3 +220,203 @@ with [`Component.render()`](#TODO) and inserting them into larger structures.
     - Directly passing rendered HTML to [`render_dependencies()`](#TODO)
 3. If you pre-render one component to pass it into another, the pre-rendered component must be rendered with
```

Comment:
Rest of the page needs cleaning up, but I will do so in 3 PR from now.

There, I implemented [this](https://github.com/django-components/django-components/issues/968#issuecomment-2653965260) and was able to remove the middleware, and also replaced `render_dependcies=False` with a separate value for `deps_strategy`.